recipes/imagemagick/all/conanfile.py

```diff
@@ -99,337 +142,257 @@ def config_options(self):
-            "--with-tiff={}".format(yes_no(self.options.with_tiff)),
-            "--with-webp={}".format(yes_no(self.options.with_webp)),
-            "--with-xml={}".format(yes_no(self.options.with_xml2)),
-            "--with-freetype={}".format(yes_no(self.options.with_freetype)),
-            "--with-djvu={}".format(yes_no(self.options.with_djvu)),
-            "--with-utilities={}".format(yes_no(self.options.utilities)),
+            f"--enable-shared={format(yes_no(self.options.shared))}",
+            f"--enable-static={format(yes_no(not self.options.shared))}",
+            f"--enable-hdri={format(yes_no(self.options.hdri))}",
+            f"--with-quantum-depth={format(self.options.quantum_depth)}",
+            f"--with-zlib={format(yes_no(self.options.with_zlib))}",
+            f"--with-bzlib={format(yes_no(self.options.with_bzlib))}",
+            f"--with-lzma={format(yes_no(self.options.with_lzma))}",
+            f"--with-lcms={format(yes_no(self.options.with_lcms))}",
+            f"--with-openexr={format(yes_no(self.options.with_openexr))}",
+            f"--with-heic={format(yes_no(self.options.with_heic))}",
+            f"--with-jbig={format(yes_no(self.options.with_jbig))}",
+            f"--with-jpeg={format(yes_no(self.options.with_jpeg))}",
+            f"--with-openjp2={format(yes_no(self.options.with_openjp2))}",
+            f"--with-pango={format(yes_no(self.options.with_pango))}",
+            f"--with-png={format(yes_no(self.options.with_png))}",
+            f"--with-tiff={format(yes_no(self.options.with_tiff))}",
+            f"--with-webp={format(yes_no(self.options.with_webp))}",
+            f"--with-xml={format(yes_no(self.options.with_xml2))}",
+            f"--with-freetype={format(yes_no(self.options.with_freetype))}",
+            f"--with-djvu={format(yes_no(self.options.with_djvu))}",
+            f"--with-utilities={format(yes_no(self.options.utilities))}",
+            f"--with-zstd={format(yes_no(self.options.with_zstd))}",
+            f"--with-fftw={format(yes_no(self.options.with_fftw))}",
+            f"--with-rsvg={format(yes_no(self.options.with_rsvg))}",
+            f"--with-autotrace={format(yes_no(self.options.with_autotrace))}",
+            f"--with-dps={format(yes_no(self.options.with_dps))}",
+            f"--with-flif={format(yes_no(self.options.with_flif))}",
+            f"--with-fpx={format(yes_no(self.options.with_fpx))}",
+            f"--with-gslib={format(yes_no(self.options.with_gslib))}",
+            f"--with-gvc={format(yes_no(self.options.with_gvc))}",
+            f"--with-jxl={format(yes_no(self.options.with_jxl))}",
+            f"--with-lqr={format(yes_no(self.options.with_lqr))}",
+            f"--with-raqm={format(yes_no(self.options.with_raqm))}",
+            f"--with-wmf={format(yes_no(self.options.with_wmf))}",
+            f"--with-raw={format(yes_no(self.options.with_raw))}",
+            "--with-gdi32=no"  # FIXME
```

Comment:
FIXME? Please, add a comment what should be fixed.

Reply:
the problem was that the file that uses gdi32 is a c++ file that uses gdiplus header, which is a c++ header. I fixed the issue by changing the extension to cpp, to indicate to msvc that it should be compiled as a c++ file.